### PR TITLE
Fuse reshapes across concat

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

**Comment by pfultz2** [comment]
More tests are needed for this.

File: test/simplify_reshapes_test.cpp [hunk 1]
```diff
@@ -1193,6 +1193,128 @@ TEST_CASE(concat_transpose4)
...
+        auto concat =
+            m1.add_instruction(migraphx::make_op("concat", {{"axis", 0}}), xreshape, yreshape);
+        m1.add_return({concat});
+    }
+    migraphx::module m2;
+    {
+        auto x      = m2.add_parameter("x", s);
+        auto y      = m2.add_parameter("y", s);
+        auto concat = m2.add_instruction(migraphx::make_op("concat", {{"axis", 0}}), x, y);
+        // TODO: This could just be a broadcast
```
Comment (TedThemistokleous) [inline test/simplify_reshapes_test.cpp:1302]: Make this a separate issue and queue it up if this a todo if it offers a perf benifit or not.

**Comment by TedThemistokleous** [comment]
rekicking CI, having another odd timeout. Looks good btw.

**Review by TedThemistokleous — APPROVED** [review]

File: src/simplify_reshapes.cpp [hunk 2]
```diff
@@ -462,6 +462,74 @@ struct find_concat_transpose
...
+
+    void apply(module& m, const match::matcher_result& mr) const
+    {
+        auto ins          = mr.result;
+        auto concat_shape = ins->get_shape();
+        auto reshapes     = ins->inputs();
+        if(reshapes.empty())
+            return;
+        auto input_shape = reshapes.front()->inputs().front()->get_shape();
+        // All inputs should have the same dimensions
```
[the remark this answers is not in the document]
Comment (pfultz2) [reply src/simplify_reshapes.cpp:481]: Yea we could, but it might require so more checking.

[the remark this answers is not in the document]
Comment (shivadbhavsar) [reply src/simplify_reshapes.cpp:481]: ok ya, we can come back to it

**Review by shivadbhavsar — APPROVED** [review]

**Pull request merged**